pallets/market/src/lib.rs

```diff
@@ -1182,6 +1279,42 @@ pub mod pallet {
             Ok((valid_deals, total_provider_lockup))
         }
 
+        /// Validates that proposals fall between the parameters set by the storage provider.
+        /// Checks for deal duration and minimum price.
+        /// returns Ok(()) if the proposal is valid, `Err(InvalidDealParameters)` if invalid.
+        fn validate_deal_parameters(
+            provider: &T::AccountId,
+            proposal: &DealProposal<T::AccountId, BalanceOf<T>, BlockNumberFor<T>>,
+        ) -> DispatchResult {
+            if let Some(params) = SPDealParameters::<T>::get(provider) {
```

Comment:
Reducing nesting here is free & makes the code easier to read 
```suggestion
            let Some(params) = SPDealParameters::<T>::get(provider) else { return Ok(()) };
```